pyxem/signals/diffraction2d.py

```diff
@@ -1981,59 +1981,38 @@ def get_azimuthal_integral1d(
         >>> det = Detector(pixel1=1e-4, pixel2=1e-4)
         >>> ds.get_azimuthal_integral1d(npt=100, detector_dist=.2, detector= det, wavelength=2.508e-12)
         """
-        if lazy_result is None:
-            lazy_result = self._lazy
-        signal_type = self._signal_type
-        unit = to_unit(self.unit)
         sig_shape = self.axes_manager.signal_shape
+        unit = self.unit
         if radial_range is None:
-            radial_range = _get_radial_extent(ai=self.ai, shape=sig_shape, unit=unit)
+            radial_range = _get_radial_extent(
+                ai=self.ai, shape=sig_shape, unit=self.unit
+            )
             radial_range[0] = 0
-
-        data_dask_array = _get_dask_array(self)
-        chunks = data_dask_array.chunks[:-2] + ((npt,),)
-        drop_axis = (len(self.axes_manager.shape) - 2, len(self.axes_manager.shape) - 1)
-        new_axis = self.axes_manager.navigation_dimension
-        integration_dask_array = _process_dask_array(
-            data_dask_array,
+        integration = self.map(
             azimuthal_integrate1d,
-            drop_axis=drop_axis,
-            new_axis=new_axis,
-            chunks=chunks,
-            output_signal_size=(npt,),
             azimuthal_integrator=self.ai,
             npt_rad=npt,
             azimuth_range=azimuth_range,
             radial_range=radial_range,
             method=method,
-            unit=unit,
+            inplace=inplace,
+            unit=self.unit,
+            mask=mask,
             sum=sum,
-            dtype=np.float32,  # pyFAI does the calculation in float32
             **kwargs,
         )
 
         # Dealing with axis changes
         if inplace:
-            result = self
-            result.axes_manager.remove(self.axes_manager.signal_axes[0])
-            result.data = integration_dask_array
-            result._lazy = True
+            k_axis = self.axes_manager.signal_axes[0]
         else:
-            result = LazySignal1D(integration_dask_array)
-            result.set_signal_type(signal_type)
-            transfer_navigation_axes(result, self)
-        k_axis = result.axes_manager.signal_axes[0]
+            transfer_navigation_axes(integration, self)
```

Comment:
I expect that this is not necessary anymore since `map` is already dealing, can it be removed?

Reply:
I think the `transfer_navigation_axes` line can be removed? At least it doesn't seem necessary in `get_azimuthal_integral2d`